contracts/rosca.sol

```diff
@@ -1,266 +1,258 @@
-
-    }
-    mapping(address => USER) members; //using struct USER to keep track of contributions and paid, allowance and etc.
-    address[] memberAddress;    //this is the only way to iterate through all the member's address
-
-    mapping(address => bool) pendingJoinRequest; // this way , address can be used as index, if we use address[] , we'll have to go thru a whole array
-
-    //deposit and withdrawal related
-    uint totalAmountInContract;
-    //bidding related state variable
-    uint lowestBid;
-    address winnerAddress;
-
-
-    modifier onlyForeman {
-        if(msg.sender != foreman)
-        throw;
-        _;
-    }
-    modifier beforeStart {
-        if(currentRound == 0)
-        throw;
-        _;
-    }
-
-    /**
-     * Creates a new ROSCA in which every roundPeriod each person contributes
-     * roundSum, for numRounds
-     */
-    function rosca(
-        uint _roundPeriodInDays,
-        uint _roundSum, //i am assuming this is referring to number epochs to run
-        uint _numEpochs,
-        int8 _minParticipants,
-        uint _startTime,
-        int16 _feeInThousandths,
-        address _feeAddress)
-    {
-        roundPeriodInDays = _roundPeriodInDays;
-        if(roundSum < 1) throw; //if everyone is only required to contribute 0 , whats the point of rosca
-        roundSum = _roundSum;
-        if(_numEpochs < 1) throw;
-        numEpochs = _numEpochs;
-        if(_minParticipants < 2) throw;// there should be at least 2 people to make a group
-        minParticipants = _minParticipants;
-        if(_startTime < now) throw;
-        startTime = _startTime;
-        if(_feeInThousandths < 0) throw; //fee must be non-negative
-        feeToService = _feeInThousandths;
-        if(feeAddress != 0) //feeAddress shouldnt be empty(null)
-        feeAddress = _feeAddress;
-
-
-        //register foreman as a member of ROSCA and increment numRounds in a Epoch to 1
-        members[msg.sender] = USER({paid: 0 , contributed: 0, alive: true, allowance: 0});
-        memberAddress.push(msg.sender);
-        numRounds = 1;
-    }
-
-
-
-    function startRound()
-    {
-
-        if(now < startTime + ( currentRound * currentEpoch  * (roundPeriodInDays * 1 days)))
-            throw;
-        else
+  function ROSCA(
+    uint16 roundPeriodInDays_,
+    uint contributionSize_, //i am assuming this is referring to number epochs to run
+    uint16 numEpochs_,
+    uint8 minParticipants_,
+    uint startTime_,
+    uint16 feeInThousandths_,
+    address feeAddress_)
+  {
+    roundPeriodInDays = roundPeriodInDays_;
+    if(contributionSize_ < MIN_CONTRIBUTION_SIZE) throw; //if everyone is only required to contribute 0 , whats the point of rosca
+    contributionSize = contributionSize_;
+
+    if(numEpochs_ < 1) throw;
+    numEpochs = numEpochs_;
+
+    if(minParticipants_ < 2) throw;// there should be at least 2 people to make a group
+    minParticipants = minParticipants_;
+
+    if(startTime_ < (now + 1 days)) throw;
```

Comment:
I suggest we try to avoid "magic numbers" in the code like "1 days" - making them constants should not change the code output size

http://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad